tests/circuit_compilation/test_code_switching_compilation.py

```
@@ -0,0 +1,122 @@
+    # Case 1: Short idle (less than SWITCHING_LENGTH=2) -> 0 bonus
+    # depths 0 and 2 implies gap of 1 (depth 1 is empty)
+    bonus = simple_graph.compute_idle_bonus(previous_depth=0, current_depth=2)
+    assert bonus == 0.0
+
+    # Case 2: Long idle
+    # depths 0 and 10 implies gap of 9
+    bonus = simple_graph.compute_idle_bonus(previous_depth=0, current_depth=10)
+    assert bonus > 0
+
+    # Ensure capacity reduction logic works
+    cap = simple_graph._edge_capacity_with_idle_bonus([0, 10], base_capacity=1.0)  # noqa: SLF001
```

Ruff (0.14.5)

Unused `noqa` directive (non-enabled: `SLF001`)

Remove unused `noqa` directive

(RUF100)